lib/Doctrine/DBAL/Connection.php

```diff
@@ -425,9 +425,46 @@ private function getDatabasePlatformVersion()
 
         // If not connected, we need to connect now to determine the platform version.
         if (null === $this->_conn) {
-            $this->connect();
+            try {
+                $this->connect();
+            } catch (\Exception $originalException) {
+                if (empty($this->_params['dbname'])) {
+                    throw $originalException;
+                }
+
+                // The database to connect to might not yet exist.
+                // Retry detection without database name connection parameter.
+                $databaseName = $this->_params['dbname'];
+                $this->_params['dbname'] = null;
+
+                try {
+                    $this->connect();
+                } catch (\Exception $fallbackException) {
+                    // Either the platform does not support database-less connections
+                    // or something else went wrong.
+                    // Reset connection parameters and rethrow the original exception.
+                    $this->_params['dbname'] = $databaseName;
```

Comment:
@mattjanssen I thought about that but as we need to backport this to `2.5` which still supports PHP `5.3`, this unfortunately is not an option